jina/serve/networking.py

```diff
@@ -707,21 +712,26 @@ def _handle_aiorpcerror(
             )
         else:
             self._logger.debug(
-                f'GRPC call failed with code {e.code()}, retry attempt {retry_i + 1}/3'
+                f'GRPC call failed with code {e.code()}, retry attempt {retry_i + 1}/{num_retries}.'
+                f' Trying next replica, if available.'
             )
 
     def _send_requests(
         self,
         requests: List[Request],
-        connection: ConnectionStubs,
+        connections: ReplicaList,
         endpoint: Optional[str] = None,
         timeout: Optional[float] = None,
     ) -> asyncio.Task:
         # this wraps the awaitable object from grpc as a coroutine so it can be used as a task
         # the grpc call function is not a coroutine but some _AioCall
         async def task_wrapper():
             metadata = (('endpoint', endpoint),) if endpoint else None
-            for i in range(3):
+            tried_addresses = set()
+            num_retries = max(3, len(connections.get_all_connections()))
+            for i in range(num_retries):
+                connection = connections.get_next_connection()
```

Comment:
I find it really confusing to have `connection` for the single entity and `connections` for the group.
Maybe `current_connexion` ?